File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Python/Exercises/18_compito_java_20170911/Iscritto.py
class Iscritto:
    def __init__(self, tipo, codice, indirizzo, nome_account, password):
        self._tipo = tipo
        self._codice = codice
        self._indirizzo = indirizzo
        self._nome_account = nome_account
        self._password = password
        
    def get_codice_iscritto(self):
        return self._codice

    def get_tipo(self):
        return self._tipo

File: II - Linguaggi di programmazione e laboratorio/Java/Exercises/18_compito_java_2017_09_11/Iscritto.java
import java.util.*;

public class Iscritto {
	protected String tipo;
	protected int codice;
	protected String indirizzo, nomeAccount, password;
	protected List<Iscritto> seguaci;

	public Iscritto(String tipo, int codice, String indirizzo, String nomeAccount, String password) {
		this.tipo = tipo;
		this.codice = codice;
		this.indirizzo = indirizzo;
		this.nomeAccount = nomeAccount;
		this.password = password;
		seguaci = new LinkedList<Iscritto>();
	}

	public void aggiungiSeguace(Iscritto i) {
		seguaci.add(i);
	}

	public int getNumeroSeguaci() {
		return seguaci.size();
	}

	public String getNome() {
		return "";
	}

	public int getId() {
		return codice;
	}

	public String toString() {
		return indirizzo + "\t" + nomeAccount + "\t" + password.length();
	}

	public String toStringFollowerNumber() {
		return "";
	}
}
